cores/esp8266/mmu_iram.h

```diff
@@ -127,73 +127,66 @@ bool mmu_is_icache(const void *addr) {
 static inline __attribute__((always_inline))
 uint8_t mmu_get_uint8(const void *p8) {
   ASSERT_RANGE_TEST_READ(p8);
-  uint32_t val = (*(uint32_t *)((uintptr_t)p8 & ~0x3));
-  uint32_t pos = ((uintptr_t)p8 & 0x3) * 8;
-  val >>= pos;
-  return (uint8_t)val;
+  return *(uint32_t *)((uintptr_t)p8 & ~0x3) >> (uintptr_t)p8 * 8;
```

Comment:
I'm not so sure about this conversion.  Before it was `>>(p8 & 3) * 3`, now it's showing `>> p8 * 8`...i.e. the bitmask was dropped.

Reply:
in GCC implementation and Xtensa ISA, rhs of shift operators is under modulo `(sizeof((lhs) + 0) * CHAR_BIT)`. and in ESP8266, `uintptr_t` and pointers are 32-bit wide,
in other words, 32-bit-lhs shift operator uses only the least significant 5 bits of rhs.

[before]
```
ptr  : 0b???????? ???????? ???????? ??????XX 
 & 3 : 0b00000000 00000000 00000000 00000011
       0b00000000 00000000 00000000 000000XX
 * 8 : 0b00000000 00000000 00000000 000XX000
                                     0bXX000
--------------------------------------------
```

[this PR]
```
ptr  : 0b???????? ???????? ???????? ??????xx
 * 8 : 0b???????? ???????? ???????? ???xx000
--------------------------------------------
                                     0bXX000
```

were your doubts cleared?

Reply:
With what I had originally, I was going for readability and ease to follow, and a little awareness of how the compiler would handle things. 

Looking closer at the changes, I just noticed the following:

This `>> (uintptr_t)p8 * 8;` doesn't look right. You want to only multiply the lower 2-bits by 8. The full multiply will exceed 32. I am not sure this would compile into something that works.  

with `p8 & ~0x3` you lose two bits of your value on shift 0.

A simple cast on the return value leaves it up to the compiler how to best reduce, bits returned, which I think compiled to an `extui`. The simple cast gives clarity and communicates the programmer's intent for the return value.

Reply:
indeed, backing to `>>(p8 & 3) * 8` brings 1 more insn: `extui a2, a2, 0, 2`.

another solution: `>>(p8 * 8 & 31)` or `>>((p8 << 3) & 31)`
this has no such drawback.

Reply:
For `>>(p8 * 8 & 31)`  the operator `&` will occur before `*`, I would expect the compiler to optimize the `8 & 31` to `8`.

I would expect `>>((p8 << 3) & 31)` to bring back the extra instruction.

Reply:
Oops sorry I got my precedences backward.

Reply:
of course, ordinary C/C++ programmers know multiplication has higher priority than bitwise-and :)

> The simple cast gives clarity and communicates the programmer's intent for the return value.

and, it's obvious what are returned from function; see prototype. abuse of explicit cast brings another kind of problem i guess.